### ISSUE TITLE
[clang-tidy] False positive for misc-unused-parameters

### ISSUE THREAD

**Issue by ishaangandhi** [llvm/llvm-project#56152]
Code in `test.H`:
```h
int myFunction(input_t level)
{
    auto orders = level.orders();
    return orders;
}

```

Command:
```console
$ clang-tidy -checks="-*,misc-unused-parameters" /path/to/test.H
```

Output:
```console
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "/path/to/test.H"
No compilation database found in /path/to or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.
1 warning and 1 error generated.
Error while processing /path/to/test.H.
/path/to/test.H:1:16: error: unknown type name 'input_t' [clang-diagnostic-error]
int myFunction(input_t level)
               ^
/path/to/test.H:1:24: warning: parameter 'level' is unused [misc-unused-parameters]
int myFunction(input_t level)
                       ^~~~~
                        /*level*/
Found compiler error(s).
```

What I would want to happen is the error to be reported, but not the warning.

Interestingly, that is what happens if you just change the function _slightly_:
```h
int myFunction(input_t level)
{
    return level.orders();
}
```

Using clang-tidy on that, you get:
```console
$ clang-tidy -checks="-*,misc-unused-parameters" /path/to/test.H
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "/path/to/test.H"
No compilation database found in /path/to or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.
1 error generated.
Error while processing /path/to/test.H.
Found compiler error(s).
/test.H:1:16: error: unknown type name 'input_t' [clang-diagnostic-error]
int myFunction(input_t level)
               ^
```

clang-tidy v13.0.0

**Comment by EugeneZelenko** [comment]
Could you please try 14 or `main`? https://godbolt.org should be helpful.

**Comment by ishaangandhi** [comment]
Exists in `clang-tidy` 15 as well:
```
[ishaangandhi@Ishaans-MBP ~/Desktop/test/failing_clangd]$ clang-tidy -checks="-*,misc-unused-parameters" test.H
1 warning and 1 error generated.
Error while processing /Users/ishaangandhi/Desktop/test/failing_clangd/test.H.
/Users/ishaangandhi/Desktop/test/failing_clangd/test.H:1:16: error: unknown type name 'input_t' [clang-diagnostic-error]
int myFunction(input_t level)
               ^
/Users/ishaangandhi/Desktop/test/failing_clangd/test.H:1:24: warning: parameter 'level' is unused [misc-unused-parameters]
int myFunction(input_t level)
                       ^~~~~
                        /*level*/
Found compiler error(s).
[ishaangandhi@Ishaans-MBP ~/Desktop/test/failing_clangd]$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 15.0.0git
  DEBUG build with assertions.
  Default target: x86_64-apple-darwin21.5.0
  Host CPU: skylake
```
